controllers/osccluster_routetable_controller_unit_test.go

```diff
@@ -0,0 +1,3292 @@
+			},
+			Subnets: []*infrastructurev1beta1.OscSubnet{
+				{
+					Name:          "test-subnet",
+					IpSubnetRange: "10.0.0.0/24",
+				},
+			},
+			InternetService: infrastructurev1beta1.OscInternetService{
+				Name: "test-internetservice",
+			},
+			NatService: infrastructurev1beta1.OscNatService{
+				Name:         "test-natservice",
+				PublicIpName: "test-publicip",
+				SubnetName:   "test-subnet",
+			},
+			RouteTables: []*infrastructurev1beta1.OscRouteTable{
+				{
+					Name:       "test-routetable",
+					SubnetName: "test-subnet",
+					Routes: []infrastructurev1beta1.OscRoute{
+						{
+							Name:        "test-route-nat",
+							TargetName:  "test-natservice",
+							TargetType:  "nat",
+							Destination: "0.0.0.0/0",
+						},
+						{
+							Name:        "test-route-igw",
+							TargetName:  "test-internetservice",
+							TargetType:  "gateway",
+							Destination: "0.0.0.0/0",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	defaultRouteTableGatewayNatReconcile = infrastructurev1beta1.OscClusterSpec{
+		Network: infrastructurev1beta1.OscNetwork{
+			Net: infrastructurev1beta1.OscNet{
+				Name:       "test-net",
+				IpRange:    "10.0.0.0/16",
+				ResourceId: "vpc-test-net",
+			},
+			Subnets: []*infrastructurev1beta1.OscSubnet{
+				{
+					Name:          "test-subnet",
+					IpSubnetRange: "10.0.0.0/24",
+					ResourceId:    "subnet-test-subnet-uid",
+				},
+			},
+			InternetService: infrastructurev1beta1.OscInternetService{
+				Name:       "test-internetservice",
+				ResourceId: "igw-test-internetservice-uid",
+			},
+			NatService: infrastructurev1beta1.OscNatService{
+				Name:         "test-natservice",
+				PublicIpName: "test-publicip",
+				SubnetName:   "test-subnet",
+				ResourceId:   "nat-test-natservice-uid",
+			},
+			RouteTables: []*infrastructurev1beta1.OscRouteTable{
+				{
+					Name:       "test-routetable",
+					SubnetName: "test-subnet",
+					ResourceId: "rtb-test-routetable-uid",
+					Routes: []infrastructurev1beta1.OscRoute{
+						{
+							Name:        "test-route-nat",
+							TargetName:  "test-natservice",
+							TargetType:  "nat",
+							Destination: "0.0.0.0/0",
+						},
+						{
+							Name:        "test-route-igw",
+							TargetName:  "test-internetservice",
+							TargetType:  "gateway",
+							Destination: "0.0.0.0/0",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+// TestGettRouteTableResourceId has several tests to cover the code of the function getRouteTableResourceId
+func TestGetRouteTableResourceId(t *testing.T) {
+	routeTableTestCases := []struct {
+		name                          string
+		spec                          infrastructurev1beta1.OscClusterSpec
+		expRouteTablesFound           bool
+		expGetRouteTableResourceIdErr error
+	}{
+		{
+			name:                          "get RouteTableId",
+			spec:                          defaultRouteTableGatewayInitialize,
+			expRouteTablesFound:           true,
+			expGetRouteTableResourceIdErr: nil,
+		},
```

Comment:
Would it be hard to add the logic of the test here ? (like [this](https://github.com/outscale-dev/osc-bsu-csi-driver/blob/5e8105db83400ae45a7ef360dc219a96db7cb426/pkg/driver/node_test.go#L678))